HEC.FDA.Model/paireddata/CurveMetaData.cs

```diff
@@ -63,69 +51,76 @@ public CurveMetaData(string xlabel, string ylabel, string name, string damageCat
         }
         public CurveMetaData(string xlabel, string ylabel, string name)
         {
-            CurveType = CurveTypesEnum.StrictlyMonotonicallyIncreasing;
             XLabel = xlabel;
             YLabel = ylabel;
             Name = name;
             DamageCategory = "unassigned";
             AssetCategory = "unassigned";
             IsNull = false;
         }
-        public CurveMetaData(string xlabel, string ylabel, string name, string damageCategory, CurveTypesEnum curveType, string assetCategory = "unassigned")
+        public CurveMetaData(string xlabel, string ylabel, string name, string damageCategory, int impactAreaID, string assetCategory = "unassigned")
         {
-            CurveType = curveType;
             XLabel = xlabel;
             YLabel = ylabel;
             Name = name;
             DamageCategory = damageCategory;
             AssetCategory = assetCategory;
             IsNull = false;
-        }
-        public CurveMetaData(string xlabel, string ylabel, string name, CurveTypesEnum curveType)
-        {
-            CurveType = curveType;
-            XLabel = xlabel;
-            YLabel = ylabel;
-            Name = name;
-            DamageCategory = "unassigned";
-            AssetCategory = "unassigned";
-            IsNull = false;
-        }
-        public CurveMetaData(string xlabel, string ylabel, string name, string damageCategory, CurveTypesEnum curveType, int impactAreaID, string assetCategory = "unassigned")
-        {
-            CurveType = curveType;
-            XLabel = xlabel;
-            YLabel = ylabel;
-            Name = name;
-            DamageCategory = damageCategory;
-            AssetCategory = assetCategory;
-            IsNull = false;
-            _ImpactAreaID = impactAreaID;
+            ImpactAreaID = impactAreaID;
         }
         public XElement WriteToXML()
         {
-            XElement masterElement = new XElement("Curve_Metadata");
-            masterElement.SetAttributeValue("CurveType", Convert.ToString(CurveType));
-            masterElement.SetAttributeValue("XLabel", XLabel);
-            masterElement.SetAttributeValue("YLabel", YLabel);
-            masterElement.SetAttributeValue("Name", Name);
-            masterElement.SetAttributeValue("DamageCategory", DamageCategory);
-            masterElement.SetAttributeValue("AssetCategory", AssetCategory);
-            masterElement.SetAttributeValue("IsNull", IsNull);
+            StoredPropertyAttribute attribute = (StoredPropertyAttribute)Attribute.GetCustomAttribute(typeof(CurveMetaData), typeof(StoredPropertyAttribute));
+            string thresholdMasterTag = attribute.SerializedName;
+
+            XElement masterElement = new XElement(thresholdMasterTag);
+
+            string xLabelTag = Serialization.GetXMLTagFromProperty(GetType(), nameof(XLabel));
```

Comment:
see other comment about using GetType() over and over.

Reply:
done